proposals/0010-stream-type.md

```diff
@@ -0,0 +1,162 @@
+- Feature Name: Stream Type
+- Start Date: 2023-03-10
+- PR: [video-dev/media-ui-extensions#10](https://github.com/video-dev/media-ui-extensions/pull/10)
+
+# Summary
+
+This is a proposal to provide well defined behaviors and an API for distinguishing between "live" content vs. "on demand" (or pre-recorded).
+
+# Motivation
+
+The idea of different “stream types” has been around for a long time in various HTTP Adaptive Streaming (HAS) standards and its precursors in some manner - minimally distinguishing between “live” content and “video on demand” / "on demand" content. However, these categories aren’t consistently named or distinguished in the same way across the various specifications. Moreover, there is no corresponding API in the browser. Yet these categories directly inform how one expects users to consume and interact with the media, including what sort of UI or “chrome” should be made available for the user. By way of example, the built in controls/UI in Safari that show up for a live src are different than those that show up for a VOD playlist. This proposal aims to normalize the names and definitions of stream types (in a way that is extensible and evolvable over time) by way of how they are expected to be consumed and interacted with by a viewer/user. It also provides a concise and easy to understand differentiator for anyone implementing different UIs/controls/"chromes" for the various stream types.
+
+An additional goal of this proposal is to recommend for MSE-based players or “playback engines” to try to normalize their use of existing APIs to be as consistent as possible with the proposed inferred stream type Algorithm.
+
+# Guide-level Explanation
+
+## Definitions
+
+- __Live Media Stream__ - Media content that is being made available in real-time at the time of playback, and the end of the media (if there is an end), will only be available sometime in the future.
+  - __NOTE__ - This definition also applies to "DVR" or "sliding window" content, where the content is being published/provided in real-time but earlier content is still also available.
```

Comment:
Should this mention that "DVR" has already finished broadcasting is likely "On Demand"/VOD at that point?

Reply:
Maybe something like this:
```suggestion
  - __NOTE__ - This definition also applies to "DVR" or "sliding window" content while the content is being published/provided in real-time but earlier content is still also available. Once it has finished broadcasting it is considered On Demand.
```

Reply:
Also, I guess https://github.com/video-dev/media-ui-extensions/pull/10#discussion_r1137842438 means that this probably shouldn't be changed.